test/Fencer/Rules/Test.hs

```diff
@@ -181,6 +149,123 @@ test_rulesLoadRulesMinimal =
       (#files [("min.yaml", minimalDomainText)] )
       (#result $ Right [minimalDomain])
 
+-- | test that 'loadRulesFromDirectory' loads a configuration file in
+-- presence of another configuration file without read permissions.
+--
+-- This matches the behavior of @lyft/ratelimit@.
+test_rulesLoadRulesReadPermissions :: TestTree
+test_rulesLoadRulesReadPermissions =
+  testCase "Configuration file read permissions" $
+    expectLoadRulesWithPermissions
+      (#ignoreDotFiles False)
+      (#files
+        [ ("domain1" </> "config.yml", domain1Text, const Dir.emptyPermissions)
+        , ("domain2" </> "config" </> "config.yml", domain2Text, id) ]
+      )
+      (#result $ Right [domain2])
+
+----------------------------------------------------------------------------
+-- Helpers
+----------------------------------------------------------------------------
+
+-- | Get a list of values on the Left or an empty list if it is a
+-- Right value.
+toErrorList :: Either [a] [b] -> [a]
+toErrorList (Right _) = []
+toErrorList (Left xs) = xs
+
+-- | Write contents to a path in the given root and modify file
+-- permissions.
+writeFile
```

Comment:
This clashes with `writeFile` in Prelude.

Reply:
It clashes if the path is not specified on usage sites. I have specified it wherever it is used.

What name would you suggested instead?

Reply:
I'll go with `writeContentsToFile` instead.

Reply:
I'd make it something like `writeFileUpdatePermissions`, which is rather verbose, but to the point.